### PR TITLE
python: Cleanups and up/downgrade

### PR DESCRIPTION
**Pull request by phlax**
- ,make `ignore_root_user_error` configurable (and remove for envoy)
- upgrade python -> 3.10.9
- downgrade `rules_python` (there seems to be a bug in latest version with pulling the toolchain)

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: bazel/repositories_extra.bzl [hunk 1]
```diff
@@ -17,7 +17,6 @@ def envoy_dependencies_extra(python_version = PYTHON_VERSION):
     python_register_toolchains(
         name = "python%s" % ("_".join(python_version.split(".")[:-1])),
         python_version = python_version,
-        ignore_root_user_error = True,
```
Comment (phlax) [inline bazel/repositories_extra.bzl:20]: i guess this could be made an arg/option of `envoy_dependencies_extra`

**Comment by phlax** [comment]
cc @keith this has exposed the lurking `cmake` issue - the vpp build is kinda doing its own thing, and not building using foreign_cc properly so it tries to use the host `cmake`

i think this needs to be addressed anyway, hopefully we can fix this without having to revert the foreign_cc update

**Review by keith — DISMISSED** [review]

**Comment by keith** [comment]
yea that wasn't even the one i hit, i think the one i hit was some contrib extension, datadog maybe

**Comment by phlax** [comment]
vpp issue is here https://github.com/envoyproxy/envoy/issues/27236

**Comment by phlax** [comment]
```console
$ git grep cmake contrib
...
```

seems to suggest that its only an issue with the vpp build, in contrib at least

**Comment by keith** [comment]
The issue was in an upstream build file

**Comment by phlax** [comment]
> The issue was in an upstream build file

i guess the problem has either gone away or essentially its a non-hermetic build issue - i can run a full rebuild which should rinse out any others

i think in this case it got triggered because i changed the python toolchain

**Comment by keith** [comment]
https://github.com/opentracing/opentracing-cpp/blob/06b57f48ded1fa3bdd3d4346f6ef29e40e08eaf5/BUILD.bazel#L36

**Comment by phlax** [comment]
nightmare 8/

not sure how we can work around this - in the case of vpp it _should_ be a foreign cc build - so altho not trivial the fix is kinda obvious

with opentracing its just another reason to be glad its going, but a bit of a blocker

**Comment by phlax** [comment]
any ideas how we can get the `cmake` bin out of the toolchain - i tried various ways, but struggled to get it working

that would provide an avenue of a hack

**Comment by phlax** [comment]
> https://github.com/opentracing/opentracing-cpp/blob/06b57f48ded1fa3bdd3d4346f6ef29e40e08eaf5/BUILD.bazel#L36


https://github.com/envoyproxy/envoy/blob/f88e29647931c7c1f1d2ff181e4d95602b0928fa/bazel/external/BUILD#L11-L12

seems like this is probably a non-issue

**Comment by phlax** [comment]
sadly is an issue 8/

```console
INFO: From Executing genrule @io_opentracing_cpp//:generate_version_h:
...
```

that also explains a few more things actually - the sooner we get rid of this the better

**Comment by htuch** [comment]
Is the rules_python downgrade related to the other changes inherently? If not, probably better as an independent patch.

**Comment by phlax** [comment]
> Is the rules_python downgrade related to the other changes inherently? 

yeah if anything upgrading python is the odd one out

the other changes were both made with the intention of getting bazel to use the cache artefact correctly

**Review by htuch — APPROVED** [review]

**Pull request merged**